test/e2e/mount_test.go

```diff
@@ -0,0 +1,190 @@
+package test
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+const zone = "us-central1-c" // matches zone in test-cluster.yaml.tmpl
+const testProjectId = "test-csi-test-infra"
+
+type testFixture struct {
+	tempDir           string
+	gcpProviderBranch string
+	testClusterName   string
+	testSecretId      string
+	kubeconfigFile    string
+}
+
+var f testFixture
+
+// Panics with the provided error if it is not nil.
+func check(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Prints and executes a command.
+func execCmd(command *exec.Cmd) error {
+	var stdout, stderr bytes.Buffer
+	fmt.Println("+", command)
+	command.Stdout = &stdout
+	command.Stderr = &stderr
+	err := command.Run()
+	fmt.Println("Stdout:", string(stdout.Bytes()))
+	fmt.Println("Stderr:", string(stderr.Bytes()))
+	return err
+}
+
+// Replaces variables in an input template file and writes the result to an
+// output file.
+func replaceTemplate(templateFile string, destFile string) error {
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	templateBytes, err := ioutil.ReadFile(filepath.Join(pwd, templateFile))
+	if err != nil {
+		return err
+	}
+	template := string(templateBytes)
+	template = strings.ReplaceAll(template, "$PROJECT_ID", testProjectId)
+	template = strings.ReplaceAll(template, "$CLUSTER_NAME", f.testClusterName)
+	template = strings.ReplaceAll(template, "$TEST_SECRET_ID", f.testSecretId)
+	template = strings.ReplaceAll(template, "$GCP_PROVIDER_BRANCH", f.gcpProviderBranch)
+	return ioutil.WriteFile(destFile, []byte(template), 0644)
+}
+
+// Executed before any tests are run. Setup is only run once for all tests in the suite.
+func setupTestSuite() {
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	f.gcpProviderBranch = os.Getenv("GCP_PROVIDER_BRANCH")
```

Comment:
Lets also get `testProjectId` and `zone` as args instead of consts

Reply:
I changed testProjectId to be an arg, but left zone and instead changed the test cluster yaml to take zone as a template arg since the test cluster doesn't have to be in the same zone as the management cluster